xmtp/src/storage/errors.rs

```diff
@@ -0,0 +1,19 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum StorageError {
+    #[error("Diesel connection error")]
+    DieselConnectError(#[from] diesel::ConnectionError),
+    #[error("Diesel result error")]
+    DieselResultError(#[from] diesel::result::Error),
+    #[error("Pool error {0}")]
+    PoolError(String),
+    #[error("Either incorrect encryptionkey or file is not a db {0}")]
+    DbInitError(String),
+    #[error("Store Error")]
+    Store(String),
+    #[error(transparent)]
+    ImplementationError(#[from] anyhow::Error),
```

Comment:
My understanding was that You should use either anyhow or thiserror. Does it make sense to have both libraries long term or is the plan to move towards 1 or the other?

Reply:
I actually got this idea from the `thiserror` [docs](https://docs.rs/thiserror/latest/thiserror/).

Reply:
heh. Learning never stops